pkg/server/server.go

```diff
@@ -172,24 +173,76 @@ func (s *Server) Run(ctx context.Context) error {
 				return nil // don't klog.Fatal. This only happens when context is cancelled.
 			}
 			klog.Infof("Bootstrapped root workspace phase 0")
-		}
 
```

Comment:
now the flow is:
1. common
2. root
3. non-root
4. common
5. root (moving 4 and 5 breaks the server, not sure it can be done in any other way)